app/assets/javascripts/discourse/app/components/post-text-selection.gjs

```diff
@@ -176,19 +175,24 @@ export default class PostTextSelection extends Component {
     quoteState.selected(postId, _selectedText, opts);
 
     let supportsFastEdit = this.canEditPost;
-    if (this.canEditPost) {
+
+    const start = getElement(selection.getRangeAt(0).startContainer);
+
+    if (!start || start.closest(CSS_TO_DISABLE_FAST_EDIT)) {
```

Comment:
This is where we actually check whether we should open the fast edit modal or the composer depending on **where** the selection is.